### PR TITLE
Fix transliterator fallback loading

### PR DESCRIPTION
**Pull request by robertbastian**
#5468

### CODE REVIEW

File: components/experimental/src/transliterate/transliterator/mod.rs [hunk 1]
```diff
@@ -352,36 +352,40 @@ impl Transliterator {
...
-            "any-hex/plain" => Ok(InternalTransliterator::Hex(
+            ))),
+            "any-hex/plain" => Some(Ok(InternalTransliterator::Hex(
                 hardcoded::HexTransliterator::new("", "", 4, Case::Upper),
-            )),
-            s => Err(DataError::custom("unavailable transliterator").with_debug_context(s)),
+            ))),
+            _ => None,
         }
     }
```
Comment (sffc) [inline components/experimental/src/transliterate/transliterator/mod.rs:391]: Suggestion (optional): invert `load_with_override` to also return `Option<Result>`

File: components/experimental/src/transliterate/transliterator/mod.rs [hunk 2]
```diff
@@ -309,7 +309,7 @@ impl Transliterator {
                 let internal_t = dep
                     // a) hardcoded specials
                     .strip_prefix("x-")
-                    .map(|special| Transliterator::load_special(special, normalizer_provider))
+                    .and_then(|special| Transliterator::load_special(special, normalizer_provider))
                     // b) the user-provided override
                     .or_else(|| Transliterator::load_with_override(&dep, lookup?).transpose())
```
Comment (sffc) [inline components/experimental/src/transliterate/transliterator/mod.rs:314]: Following the logic here:

1. `dep.strip_prefix` returns `Option<...>`
2. `.and_then` now resolves to `Option<Result<...>>`. Will be `None` if either `strip_prefix` failed or if `load_special` did not match anything. OK so far.
3. `.or_else` is skipped if `load_special` returned `Some`. Otherwise, it returns `Option<Result<...>>`, with a `None` value if there was no override. Still OK.
4. `.unwrap_or_else` returns the `Result<...>` or else calls `load_rbt` and propagates the result from that.

Previously, the behavior was:

1. `dep.strip_prefix` returns `Option<...>`
2. `.map` resolves to `Option<Result<...>>`. Will be `None` _only if_ strip_prefix failed.
3. …

So the difference is that we never previously called `load_with_override` or `load_rbt` if the ID started with the string `x-`. As far as I can tell, the `x-` gets added here:

https://github.com/unicode-org/icu4x/blob/287e3ff0a3598ce57926664e23c5883b2d9a4d71/components/experimental/src/transliterate/compile/pass2.rs#L397

Note that `load_with_override` already fails unconditionally on `x-` because it needs the ID to be valid BCP-47. However, `load_rbt` just takes the ID and stuffs it into the DataMarkerAttributes.

Basically, I think this means that a string referencing a transliterator ID that doesn't have a BCP-47 alias, such as "::Lower" (which appears to become "x-any-lower"), could now be returned by a data provider that responds to requests for "und/x-any-lower" (using "/" as the data marker delimiter). So, that is the nature of the behavior change.

Without this change, it is _required_ to always add a BCP-47 alias mappings for any transliterator IDs that are not in the built-in set.

**Review by sffc — COMMENTED** [review]
I left some commentary

**Review by Manishearth — APPROVED** [review]

**Review by sffc — DISMISSED** [review]
OK. Were you comfortable with the observations from the comments I left above?

**Comment by robertbastian** [comment]
Yes

**Review by sffc — APPROVED** [review]
